source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRHostConfig.mm

```diff
@@ -0,0 +1,144 @@
+        case TextColor::Good: {
+            str = self.m_hostConfig->colors.good.normal;
+            break;
+        }
+        case TextColor::Warning: {
+            str = self.m_hostConfig->colors.warning.normal;
+            break;
+        }
+        case TextColor::Attention: {
+            str = self.m_hostConfig->colors.attention.normal;
+            break;
+        }
+        default: {
+            str = self.m_hostConfig->colors.good.normal;
+            break;
+        }
+    }
+    
+    num = [self retrieveIntFromHostConfigString:[[NSString alloc] initWithCString:str.c_str()
+                                encoding:[NSString defaultCStringEncoding]]];
+
+    return [UIColor colorWithRed:((num & 0x00FF0000) >> 16) / 255
+                           green:((num & 0x0000FF00) >> 8)  / 255
+                            blue:((num & 0x000000FF))       / 255
+                           alpha:((num & 0xFF000000) >> 24) / 255];
+
+}
+
+- (CGSize) getImageSize:(std::shared_ptr<Image>) imgBlock {
+    float sz = self.m_hostConfig->imageSizes.mediumSize;
+    switch (imgBlock->GetImageSize()){
+        case ImageSize::Large: {
+            sz = self.m_hostConfig->imageSizes.largeSize;
+            break;
+        }
+        case ImageSize::Medium: {
+            sz = self.m_hostConfig->imageSizes.mediumSize;
+            break;
+        }
+
+        case ImageSize::Small: {
+            sz = self.m_hostConfig->imageSizes.smallSize;
+            break;
+        }
+
+        default: {
+            NSLog(@"unimplemented");
+        }
+    }
+    CGSize cgSize = CGSizeMake(sz, sz);
+    return cgSize;
+}
+
+- (int) getTextBlockTexSize:(std::shared_ptr<TextBlock>)txtBlock {
+    switch (txtBlock->GetTextSize()){
+        case TextSize::Small:
+            return self.m_hostConfig->fontSizes.smallFontSize;
+        case TextSize::Normal:
+            return self.m_hostConfig->fontSizes.normalFontSize;
+        case TextSize::Medium:
+            return self.m_hostConfig->fontSizes.mediumFontSize;
+        case TextSize::Large:
+            return self.m_hostConfig->fontSizes.largeFontSize;
+        case TextSize::ExtraLarge:
+            return self.m_hostConfig->fontSizes.extraLargeFontSize;
+        default:
+            return self.m_hostConfig->fontSizes.normalFontSize;
+    }
+}
+    
+-(NSTextAlignment) getTextBlockAlignment:(std::shared_ptr<TextBlock>) txtBlock {
+    switch (txtBlock->GetHorizontalAlignment()){
+        case HorizontalAlignment::Center:
+            return NSTextAlignmentCenter;
+        case HorizontalAlignment::Left:
+            return NSTextAlignmentLeft;
+        case HorizontalAlignment::Right:
+            return NSTextAlignmentRight;
+        default:
+            return NSTextAlignmentLeft;
+    }
+}
+     
+-(NSNumber* ) getTextBlockTextWeight:(std::shared_ptr<TextBlock>) txtBlock{
+    switch (txtBlock->GetTextWeight()) { 
+        case TextWeight::Normal:
+            return [[NSNumber alloc] initWithInt:0];
```

Comment:
thanks

---
In reply to: [122551696](https://github.com/Microsoft/AdaptiveCards/pull/423#discussion_r122551696) [](ancestors = 122551696)